subsys/bluetooth/host/conn.c

```diff
@@ -1502,7 +1502,7 @@ struct bt_conn *conn_lookup_iso(struct bt_conn *conn)
 			return iso_conn;
 		}
 
-		if (conn->iso.acl == conn) {
+		if (iso_conn->iso.acl == conn) {
```

Comment:
I think we in general could benefit from naming all ISO bt_conn like iso_conn. It is quite confusing when reading the code what type of conn we are currently dealing with. 

```
__ASSERT(conn->type == BT_CONN_TYPE_ISO, "Invalid connection type");
```
Might also do the trick.

Reply:
I don't think we should generalize this, it quickly leads to start using notations for variables names, so I used short names whenever possible since the context should be clearer, that said we could have a function to access type specific values which can then use CHECKIF to validate the type.